### ISSUE TITLE
Recursive search inside child property of all array elements only looks at last element of array

### ISSUE THREAD

**Issue by bdcarr** [Galbar/JsonPath-PHP#64]
I'm seeing this issue with version 2.1.

(I know this repo isn't actively maintained, I just want to document the issue. There's a small chance I'll try to fix it myself but I don't really know PHP or regex very well.)

I'm working with a schema that's like this (extraneous fields excluded):

```json
{
  "success": true,
  "result": [
    {
      "data": {
        "someField": "value1",
        "nestedObj": {
            "someField": "otherValue1"
        }
      }
    },
    {
      "data": {
        "someField": "value2",
        "nestedObj": {
            "someField": "otherValue2"
        }
      }
    },
    {
      "data": {
        "someField": "value3",
        "nestedObj": {
            "someField": "otherValue3"
        }
      }
    }
  ]
}
```

To get the values of every `someField` instance, I'd expect to be able to do this: `"$.result[*].data..someField"`. But when I do that, I only get the values from the last element in the `result` array: `["value3", "otherValue3"]`.

These work:
* `"$.result[*].data.nestedObj.someField"` (doesn't do what I want, but it gets all the values it's supposed to
* `"$.result[*]..someField"` (does what I want, but less precise)
* `"$..someField"` (even less precise, a true yolo query)

Here's some PHP you can use to replicate it - you'd expect to get `["online","reference","fiction","bricks n mortar","fiction","fiction"]` but instead you get `["bricks n mortar","fiction","fiction"]`.

```php
$obj = new JsonObject(
    <<<EOT
{
  "results": [
    {
      "store": {
        "category": "online",
        "books": [
          {
            "category": "reference",
            "author": "Nigel Rees",
            "title": "Sayings of the Century",
            "price": 8.95,
            "available": true
          },
          {
            "category": "fiction",
            "author": "Evelyn Waugh",
            "title": "Sword of Honour",
            "price": 12.99,
            "available": false
          }
        ]
      }
    },
    {
      "store": {
        "category": "bricks n mortar",
        "books": [
          {
            "category": "fiction",
            "author": "Herman Melville",
            "title": "Moby Dick",
            "isbn": "0-553-21311-3",
            "price": 8.99,
            "available": true
          },
          {
            "category": "fiction",
            "author": "J. R. R. Tolkien",
            "title": "The Lord of the Rings",
            "isbn": "0-395-19395-8",
            "price": 22.99,
            "available": false
          }
        ]
      }
    }
  ]
}
EOT
);

$results = json_encode($obj->get("$.results[*].store..category"));

echo("results: $results");
```

**Comment by Galbar** [comment]
Hello @bdcarr!

Very interesting find! I just tried it and I have confirmed what you report. I have some time today and maybe tomorrow and I will be giving it a look.

I will share what I find.

**Comment by Galbar** [comment]
Hi again @bdcarr!

It has been tough, but I found it. There was a bug in the recursive operator where it would only take the last element of the result of the path leading to it. This bug made visible another bug with the recursive operator. In summary, it was broken. I have uploaded the changes in commit https://github.com/Galbar/JsonPath-PHP/commit/5ac1783ed447063357f472a080d56dbdfe5b4ee8. There is a more in-detail explanation to what I found in the commit message.

Given that this will break backwards compatibility, I want to take advantage and increase the minimum PHP supported version before releasing. I don't expect for it to be a problem but it won't be today as I now have to do other things. I'll try to do it ASAP, but can't promise anything.
